modules/cij/runner.py

```diff
@@ -886,63 +820,59 @@ def main(conf):
     trun_to_file(trun)              # Persist trun
     trun_to_junitfile(trun)         # Persist as jUNIT XML
 
-    tr_err = 0
-    tr_ent_err = trun_enter(trun)
-    for tplan in (tp for tp in trun.testplans if not tr_ent_err):
-        tp_err = 0
-        tp_ent_err = tplan_enter(trun, tplan)
+    trun.enter(trun)
 
-        for tsuite in (ts for ts in tplan.testsuites if not tp_ent_err):
-            ts_err = 0
-            ts_ent_err = tsuite_enter(trun, tsuite)
+    for tplan in (tp for tp in trun.testplans if trun.entered):
+        tplan.enter(trun)
 
-            for tcase in (tc for tc in tsuite.testcases if not ts_ent_err):
-                tc_err = 0
+        for tsuite in (ts for ts in tplan.testsuites if tplan.entered):
+            tsuite.enter(trun)
```

Comment:
We could avoid evaluating the static values `trun.entered`, `tplan.entered`, and `tsuite.entered` for every item in each list if we did something like

```python
if tplan.enter(trun):
  break
  
  for tsuite in tplan.testsuites:
    if tsuite.enter(trun):
      break
      
    for tcase in tsuite.testcases:
      ...
```
I think that this reads better for the three nested loops. But I can't come up with a good solution to skip the outer loop in the same way without wrapping the outermost loop in `if trun.entered:`, which would nest this thing even deeper

Reply:
Yeah, there are also the other side-effects at the end of the loop-nests to update run exit() and assign status.
A way to make this prettier would probably be to move the "run" into the runnable and call recursively.